sources/unstructured_data/inbox/__init__.py

```diff
@@ -0,0 +1,158 @@
+"""This source collects inbox emails, downloads attachments to local folder and stores all info in destination"""
+import os
+from copy import deepcopy
+from typing import Any, Optional, Sequence
+import email
+import imaplib
+
+
+import dlt
+from dlt.common import logger, pendulum
+from dlt.extract.source import TDataItem, DltResource
+
+from .settings import FILTER_EMAILS, STORAGE_FOLDER_PATH
+
+
+@dlt.source
+def inbox_source(
+    storage_folder_path: str = STORAGE_FOLDER_PATH,
+    filter_emails: Sequence[str] = FILTER_EMAILS,
+    attachments: bool = False,
+    start_date: pendulum.DateTime = pendulum.datetime(2000, 1, 1),
+) -> DltResource:
+    uids = messages_uids(filter_emails=filter_emails, folder="INBOX", start_date=start_date)
+
+    if attachments:
+        return uids | get_attachments_by_uid(storage_folder_path=storage_folder_path)
+    else:
+        return uids | read_messages
+
+
+@dlt.resource
+def messages_uids(
+    host: str = dlt.secrets.value,
+    email_account: str = dlt.secrets.value,
+    password: str = dlt.secrets.value,
+    filter_emails: Sequence[str] = FILTER_EMAILS,
+    folder: str = "INBOX",
+    start_date: pendulum.DateTime = pendulum.datetime(2000, 1, 1),
+    initial_message_num: Optional[Any] = dlt.sources.incremental(
+        "message_uid", initial_value=1
```

Comment:
are getting the message with uid == 1? I think we have inclusive lower bound so you should

Reply:
it is the lowest, I cannot take 0 or -1 it raises errors